### PR TITLE
Shift and rotate bitwise instructions

### PR DESCRIPTION
**Pull request by toon-leader-bacon**

### CODE REVIEW

File: src/main/java/group1/mips_simulator/components/instructionExecution/InstructionExecutions.java [hunk 1]
```diff
@@ -545,4 +546,71 @@ public ExecutionResult execute_sir(Computer computer, RXIA_Instruction i) {
...
+     * Shift Register by Count
+     * c(r) is shifted left (L/R =1) or right (L/R = 0) either logically (A/L = 1) or arithmetically (A/L = 0)
+     * XX, XXX are ignored
+     * Count = 0…15
+     * If Count = 0, no shift occurs
+     */
+    public ExecutionResult execute_src(Computer computer, Bitwise_Instruction i) {
+        Register targetReg = computer.cpu.regfile.getGPR(i.getR().value);
+
+        boolean shiftLeft = (i.getLeftRight().value == 1);
```
Comment (akm303) [inline src/main/java/group1/mips_simulator/components/instructionExecution/InstructionExecutions.java:562]: what is this function, `getLeftRight` ?
most of the other ones i kinda figure out from context clues, but i missed this one

File: src/test/java/group1/mips_simulator/components/InstructionExecutionsTest.java [hunk 2]
```diff
@@ -205,4 +206,60 @@ void execute_jz() {
...
+            } else {
+                // arithmetic
+                value = (short) (value >> count); // Unsigned
+            }
+        }
+        return value;
+    }
+
+    @Test
+    void shitTests() {
```
Comment (akm303) [inline src/test/java/group1/mips_simulator/components/InstructionExecutionsTest.java:225]: 👀👀

**Review by akm303 — APPROVED** [review]
looks good

File: src/test/java/group1/mips_simulator/components/InstructionExecutionsTest.java (hunk 2 above)
Comment (toon-leader-bacon) [reply src/test/java/group1/mips_simulator/components/InstructionExecutionsTest.java:225]: Lol oops. I meant "shift" tests. But the title is still accurate....

File: src/main/java/group1/mips_simulator/components/instructionExecution/InstructionExecutions.java (hunk 1 above)
Comment (toon-leader-bacon) [reply src/main/java/group1/mips_simulator/components/instructionExecution/InstructionExecutions.java:562]: Are we shifting left or right. It's simply called `L/R` in the documentation which is kinda confusing :/ 

If you have a better name suggestion I'm all ears

**Pull request merged**